lib/matplotlib/text.py

```diff
@@ -177,8 +177,13 @@ def __getstate__(self):
         d = super().__getstate__()
         # remove the cached _renderer (if it exists)
         d['_renderer'] = None
+        d['_cached'] = None
```

Comment:
You can instead do `d["_cached"] = {}` and then you won't need to implement setstate below.

Reply:
Thanks @anntzer. May I ask why skipping the `__setstate__` is a better approach here? Generally speaking, isn't it better to have `__getstate__` _and_ `__setstate__`? (I may be wrong)

Reply:
In general, I think it's always better if you can get away with less code (if you don't have to obfuscate anything).  Additionally, in the case here, one may consider that `_cached = None` is actually a non-valid state for the Text object (in a statically typed language, _cached would be declared as `dict`, not as `dict | None`); obviously it only exists in that non-valid state for a tiny period of time as `__getstate__` immediately readjusts that before the object is returned, but given that you can easily completely avoid ever reaching this non-valid state, I'd rather have that.

Reply:
I see, thanks for the nice explanation!